runtime/compiler/z/env/J9CPU.cpp

```diff
@@ -50,19 +50,42 @@ J9::Z::CPU::applyUserOptions()
    OMRPORT_ACCESS_FROM_OMRPORT(TR::Compiler->omrPortLib);
 
    if (_processorDescription.processor >= OMR_PROCESSOR_S390_Z10 && TR::Options::getCmdLineOptions()->getOption(TR_DisableZ10))
-      _processorDescription.processor = OMR_PROCESSOR_S390_FIRST;
+      {
+      // Z10 is the minimum supported architecture for the OpenJ9 on Z. So instead of reducing the architecture to Z9 here,
+      // we go down to Unknown.
+      _supportedArch = Unknown;
+      _processorDescription.processor = OMR_PROCESSOR_S390_UNKNOWN;
+      }
    else if (_processorDescription.processor >= OMR_PROCESSOR_S390_Z196 && TR::Options::getCmdLineOptions()->getOption(TR_DisableZ196))
+      {
+      _supportedArch = z10;
```

Comment:
`_supportedArch` is supposed to be set by the old implementation. If you are going to set `_supportedArch` using the new implementation, the tests (old implementation against new implementation) will always pass.

Reply:
please add `_supportedArch` in `J9::Z::CPU::initializeS390ProcessorFeatures()` instead of `J9::Z::CPU::applyUserOptions()`. `J9::Z::CPU::initializeS390ProcessorFeatures()` sets `_supportedArch` based on the old implementation (j9port) and `J9::Z::CPU::applyUserOptions()` sets _processorDescription based on the new implementation (omrport).

Reply:
`_supportedArch` and `J9::Z::CPU::initializeS390ProcessorFeatures()` will go away when the old processor detection implementation gets removed. But for now the best thing to do is to not invalidate the tests by applying the new changes to both old and new implementation.